substrate/frame/support/procedural/src/pallet/parse/config.rs

```diff
@@ -403,11 +448,39 @@ impl ConfigDef {
 							trait_item.span(),
 							"Invalid #[pallet::constant] in #[pallet::config], expected type item",
 						)),
+					// Pallet developer has explicitly requested to include metadata for this associated type.
+					//
+					// They must provide a type item that implements `TypeInfo`.
+					(PalletAttrType::IncludeMetadata(_), syn::TraitItem::Type(ref typ)) => {
+						if is_event {
+							return Err(syn::Error::new(
+								pallet_attr._bracket.span.join(),
+								"Invalid #[pallet::include_metadata] in `type RuntimeEvent`, \
+								expected type item. The associated type `RuntimeEvent` is already collected.",
+							))
+						}
+
+						if already_constant {
```

Comment:
So, if I put `include_metadata` before `constant`, this branch will not be taken.

Reply:
Thanks for catching this 🙏 

Have moved the checks after we iterate over all pallet attributes